commit 1cc502fc1dc46cedd20b833903eee60ee70d34f5 (refs/changes/44/3844/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-04-19T15:59:31-07:00 (1 year, 6 months ago)
    
    TRITON-329 vmapi_vms entries without a 'value.uuid' break VMAPI startup data-migration
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/apis/moray.js b/lib/apis/moray.js
index 8f3970a..34c3a77 100644
--- a/lib/apis/moray.js
+++ b/lib/apis/moray.js
@@ -1789,10 +1789,11 @@ Moray.prototype.putBatch = function putBatch(modelName, records, callback) {
     assert.string(bucketName, 'bucketName');
 
     this._morayClient.batch(records.map(function generateVmPutBatch(record) {
+        assert.string(record.key, 'record.key');
         return {
             bucket: bucketName,
             operation: 'put',
-            key: record.value.uuid,
+            key: record.key,
             value: record.value,
             etag: record._etag
         };
